### ISSUE TITLE
Displaying global options when listing options for a command.

### ISSUE THREAD

**Issue by merijn** [pcapriotti/optparse-applicative#294]
I often find myself having the same options shared across multiple/all commands (i.e. verbosity) and specify those as global options and this all works fine. However, these global options are only shown when someone runs `foo --help` and not when running `foo subcommand --help`. Personally I find it rather annoying to have to run a command/help twice to figure out all the options I can specify.

I would like some way to either 1) annotate a command so that it will repeat all global options when it's help is displayed or 2) annotate a (global) option so that it is repeated for all commands.

**Comment by HuwCampbell** [comment]
I'll have a think about it. I understand the frustration for sure.

My current thoughts are around how backtracking and "global" parser types will be inconsistent with this style of API. I can think of a few ways.

I do think there are new functions which can be written to ask for the "global options" help, and combine them with the subcommand help before pretty printing. We just need to find the right API.

**Comment by domenkozar** [comment]
Got an user that was confused about the current `subparser` --help split: https://github.com/cachix/cachix/issues/126#issuecomment-424042513

**Comment by HuwCampbell** [comment]
I have a dirty prototype version this working now.

I think for subparser inline it very much makes sense to show this information; I'm not quite so sure about the other run methods; it's probably best to not in no backtrack mode.

It's also a bit dirty with the way `--help` is implemented. As that's usually a separate parser per subcommand, so it shows up multiple times.

**Comment by HuwCampbell** [comment]
This is the basic idea https://github.com/pcapriotti/optparse-applicative/compare/topic/global-options?expand=1

**Comment by HuwCampbell** [comment]
Currently looks like this

```
Usage: cabal configure [--enable-tests] [-f|--flags FLAGS]
  Prepare to build the package

Available options:
  --enable-tests           Enable compilation of test suites
  -f,--flags FLAGS         Enable the given flag
  -h,--help                Show this help text

Global options:
  -v,--verbose LEVEL       Set verbosity to LEVEL
  --version                Print version information
  -h,--help                Show this help text
```

**Comment by domenkozar** [comment]
:tada:

**Comment by merijn** [comment]
I'll try this this weekend :)

**Comment by HuwCampbell** [comment]
Cool. I've just fixed the duplicate help issue as well.

**Comment by HuwCampbell** [comment]
Addressed in #385
